forem/forem
# Community Wellness badge

## What type of PR is this? (check all applicable)


## Description

Implements the Community Wellness Badge behind a Feature Flag

## Related Tickets & Documents

- Relates to #17310

## QA Instructions, Screenshots, Recordings

Since this is a feature that works with data that exists over time it may be tricky to test locally (check out the specs in place). However, if feeling confident about trying it out locally you can:

1. Boot app locally
2. Create comments on existing posts
3. Update the comment dates (and their corresponding reactions) using the console
   - You need at least two comments per week
   - Comments must be at least 3 days old (first week award)
   - Reaction dates may have to be updated like this as well ([spec helper takes this into account](https://github.com/forem/forem/pull/17394/files#diff-adacbae246695183a46a623a417898510a8e7282dda432c18775703e60c9798eR12))
5. Run the service using the console
   - `Badges::CommunityWellnessBadge.call`
6. Check out Ahoy tracking to see if the Badge was attempted to be awarded on your user

If you add a negative reaction on the comments (mod/trusted user) the comment shouldn't count for the badge

### UI accessibility concerns?

None - no new HTML was introduced

## Added/updated tests?


## [Forem core team only] How will this change be communicated?


## [optional] Are there any post deployment tasks we need to perform?

Keep an eye out for the next execution of the badge service (~00:45 UTC) and double check results/performance

## [optional] What gif best describes this PR or how it makes you feel?

![riding unicorn through data](https://media.giphy.com/media/l0HlN5Y28D9MzzcRy/giphy.gif)

**Above: The new service parsing through the `comments` table**
